hyperswitchSDK/Core/HyperPaymentMethodManagement/PaymentMethodManagementWidget.swift

```diff
@@ -0,0 +1,108 @@
+//  hyperswitch
+//
+//  Created by Shivam Nan on 18/10/24.
+//
+
+import Foundation
+import UIKit
+import WebKit
+
+/// PaymentSheetResult is an enum that represents the possible outcomes of a payment sheet operation.
+@frozen public enum PaymentMethodManagementResult {
+    case closed(data: String)
+    case failed(error: Error)
+}
+
+internal class PaymentMethodManagementWidget: UIControl {
+    private var ephemeralKey: String?
+    internal static var onAddPaymentMethod: (() -> Void)?
+    internal var completion: ((PaymentMethodManagementResult) -> ())?
+    
+    // Initialize the widget with the ephemeral key and configuration.
+    public init(ephemeralKey: String, frame: CGRect, onAddPaymentMethod: (() -> Void)?, completion: @escaping (PaymentMethodManagementResult) -> ()) {
+        self.ephemeralKey = ephemeralKey
+        PaymentMethodManagementWidget.onAddPaymentMethod = onAddPaymentMethod
+        self.completion = completion
+        super.init(frame: frame)
+        commonInit()
+    }
+    
+    required public init?(
+        coder: NSCoder
+    ) {
+        super.init(coder: coder)
+        commonInit()
+    }
+    
+    public override init(
+        frame: CGRect
+    ) {
+        super.init(frame: frame)
+        commonInit()
+    }
+    
+    //    public override var intrinsicContentSize: CGSize {
+    //        return CGSize(width: self.frame.width, height:  self.frame.height)
+    //    }
+    
+    private func commonInit() {
+        let hyperParams = [
+            "appId" : Bundle.main.bundleIdentifier,
+            "sdkVersion" : SDKVersion.current,
+            "country" : NSLocale.current.regionCode,
+            "ip": nil,
+            "user-agent": WKWebView().value(forKey: "userAgent"),
+            "launchTime": Int(Date().timeIntervalSince1970 * 1000)
+        ]
+        
+        // Prepare the props to send to the React Native module.
+        let props: [String : Any] = [
+            "type": "paymentMethodsManagement",
+            "hyperParams": hyperParams,
+            "customBackendUrl": APIClient.shared.customBackendUrl as Any,
+            "customParamas": APIClient.shared.customParams as Any,
+            "ephemeralKey": self.ephemeralKey ?? "",
+            "publishableKey": APIClient.shared.publishableKey as Any,
+        ]
+        
+        RNViewManager.sharedInstance.responseHandler = self
+        
+        // Get the React Native view from RNViewManager.
+        let rootView = RNViewManager.sharedInstance.viewForModule("hyperSwitch", initialProperties: ["props": props])
+        
+        rootView.frame = self.bounds
+        
+        // Add the React Native view to the current view.
+        addSubview(rootView)
+    }
+    
+    internal static func exitWidget() {
+        print("-- exit widget")
+        DispatchQueue.main.async {
+            if let view = RNViewManager.sharedInstance.rootView {
+                let reactNativeVC: UIViewController? = view.reactViewController()
```

Comment:
you are adding widget view to a UIViewController and then dismissing it, try to just remove the view